compiler/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -296,7 +296,7 @@ pub struct Contract {
 /// Given a FileId, fetch the File, from the FileManager and parse it's content
 pub fn parse_file(fm: &FileManager, file_id: FileId) -> (ParsedModule, Vec<ParserError>) {
     let file_source = fm.fetch_file(file_id).expect("File does not exist");
-    parse_program(file_source)
+    parse_program(file_id.as_usize().into(), file_source)
```

Comment:
Why don't we use `FileId`s directly instead of source ids?

Reply:
reworked to use SrcId across instead